lib/vector-core/src/fanout.rs

```diff
@@ -74,7 +74,7 @@ impl Fanout {
 
     fn remove(&mut self, id: &ComponentKey) {
         assert!(
-            self.senders.remove(id).is_some(),
+            self.senders.shift_remove(id).is_some(),
```

Comment:
Is this related to this PR?

Reply:
This is a result of bumping the `indexmap` version.